### PR TITLE
Add phc.easy.Observations class

### PR DESCRIPTION
**Pull request by rcdilorenzo**
This new easy class pulls out observations and expands all of the relevant columns. It also performs paging as needed. I'm testing everything in PHC notebooks, but I'll work in the future on separating the API calls with the transforms so I can write more unit tests. 

Next up will be a progress indicator and/or file streaming in the Query class to enable our data-intensive research needs. I was able to successfully test this code with a set of +110K observations.

### CODE REVIEW

**Comment by mschroering** [comment]
> Next up will be a progress indicator and/or file streaming in the Query class to enable our data-intensive research needs.

Can you expand on what you mean by file streaming?  The PHC has a data lake feature https://docs.us.lifeomic.com/user-guides/data-lake/ that allows one to export views of the data using SQL like queries.  The results are stored as a CSV file that can be fetched for analysis.  I just want to make sure you were aware and we aren't building something similar in nature.  @mjtieman is the Data Lake SME.

**Comment by mjtieman** [comment]
Expanding on Mark's comment. It seems like these changes overlap quiet a bit with our existing data-lake APIs. The SDK has support for running queries (sync or async), writing results to file and loading into a dataframe. The only places where the existing functionality does not overlap is the underlying service being hit and the shape of the results, i.e. results from FHIR search service contains more columns than the data-lake results.

**Comment by mjtieman** [comment]
@rcdilorenzo discussed the use of FHIR Search Service (FSS) vs the Data Lake (DL) and decided FSS is the better option. Currently, the DL only has a small subset of columns offered by FSS. Given one of the use cases for these changes is model training, more features is preferred. 

With respect to writing the dataframe to file. The file would reside on the local file system and essentially serve as a cache to be used when a notebook cell is re-run.

File: phc/easy/query.py [hunk 1]
```diff
@@ -49,43 +79,31 @@ def execute_dsl(
...
+                auth_args,
+            )
+
+        # Scroll if limit is above MAX_RESULT_SIZE
+        limit = iter(query.get("limit", []))
+
+        lower = next(limit, {}).get("value", 0)
+        upper = next(limit, {}).get("value", 0)
+
+        scroll = True if upper - lower > MAX_RESULT_SIZE else all_results
```
Comment (mjtieman) [inline phc/easy/query.py:107]: Minor: can directly assign the comparison result, i.e. `scroll = upper -  lower > MAX_RESULT_SIZE`

Comment (mjtieman) [inline phc/easy/query.py:107]: Minor: can directly assign the comparison result, i.e. `scroll = upper - lower > MAX_RESULT_SIZE`

Comment (rcdilorenzo) [reply phc/easy/query.py:107]: This should be resolved in a later PR.

**Pull request merged**